### DISCUSSION TITLE
Use vh to set element height

### DISCUSSION THREAD

**Discussion by reposman33** [tailwindlabs/tailwindcss#2123] [Help]
Hi,

I am new to Tailwind so first thing I do is see if the css rules I use are available as tailwind classes. I find it handy to set the height of a container (div) to a fraction of the viewport. eg `height: 10vh;`  but I cannot find a corresponding Tailwind class doing this.

Is there a way to achieve this out of the box? Or do I have to tweak tailwind...

Thanks!

**Comment by florianbouvot** [comment]
Hi @reposman33,

You can add your own height name/values, look at : 

https://tailwindcss.com/docs/configuration 
https://tailwindcss.com/docs/height

**Comment by reposman33** [comment]
Thanks, I got this in my tailwind.config.js:

```
	theme: {
		extend: {},
		height: {
			"10v": "10vh",
			"20v": "20vh",
			"30v": "30vh",
			"40v": "40vh",
			"50v": "50vh",
			"60v": "60vh",
			"70v": "70vh",
			"80v": "80vh",
			"90v": "90vh",
			"100v": "100vh",
		},
	},
```
I want a div to have stylerule ```height: 10vh``` so I add h-10h like this: 
```
<div id="app" class="container bg-gray-100 rounded shadow-lg h-10v mx-auto mt-20 px-4"></div>
```
The other styles get parsed but the h-10v not. Am I specifying it wrong in the config?

**Reply by florianbouvot** [reply]
@reposman33 you can add `height: { ... }` in the `extend: {}` to keep existing height.

Can you share more about your config (webpack, gulp, ...)?

Did you check your css output file: does `h-10v` exist?

**Reply by trusktr** [reply]
For reference, it works like this:

```js
	theme: {
		extend: {
			height: {
				"10v": "10vh",
				"20v": "20vh",
				"30v": "30vh",
				"40v": "40vh",
				"50v": "50vh",
				"60v": "60vh",
				"70v": "70vh",
				"80v": "80vh",
				"90v": "90vh",
				"100v": "100vh",
			},
		},
	},
```

and then `class="h-10v"` will work.

**Reply by anduscheung** [reply]
is there any option we can add {x}-vh dynamically where x can be any number?

**Reply by NovusOpus** [reply]
> is there any option we can add {x}-vh dynamically where x can be any number?

Tailwind Jit Arbitrary Value is what you're looking for: https://tailwindcss.com/docs/just-in-time-mode#arbitrary-value-support

**Comment by NovusOpus** [comment]
You can also use [Jit Arbitrary value Support](https://tailwindcss.com/docs/just-in-time-mode#arbitrary-value-support), so you can write the value in-class without having to set specific heights in the `tailwind.config` file.

Like this:
```html
<div class="h-screen w-full  bg-blue-900 flex items-center justify-center">
  <div class="w-[60%] h-[10vh] bg-sky-700">
      Some Example container
  </div>
</div>
```

Here's the [Tailwind Play Link (With Jit Enabled)](https://play.tailwindcss.com/DhTYVAZUN6) so you can play with it.
